server/routes/controllers/authentication.js

```diff
@@ -0,0 +1,68 @@
+const express = require("express");
+const db = require("../../db/models");
+//const { OAuth2Client } = require("google-auth-library");
+const { google } = require("googleapis");
+const User = require("../../db/models/User");
+const auth = require("../../middleware/auth");
+
+const router = new express.Router();
+
+const oAuthClient = new google.auth.OAuth2(
+  process.env.AUTH_CREDENTIALS,
+  process.env.AUTH_SECRET,
+  process.env.AUTH_REDIRECT_PATH
+);
+
+// login with googleAuth
+router.post("/api/authentication/google", async (req, res) => {
+  //code from google auth, email=user email, variant=type of authentication [login vs signup]
+  const { code, email, variant } = req.body;
+  oAuthClient.getToken(code, async (err, token) => {
+    if (err) {
+      console.log(err);
+      res.status(500).send("Authentication error.", err);
```

Comment:
This line is crashing the app if there's an error. Express interprets res.send(a, b) as a being the status code and b being the body, even if status code is already set.